app/controllers/glossary_terms_controller.rb

```diff
@@ -39,24 +37,30 @@ def edit
   # ---------- Actions to Modify data: (create, update, destroy, etc.) ---------
 
   def create
-    glossary_term = \
-      GlossaryTerm.new(user: @user, name: params[:glossary_term][:name],
-                       description: params[:glossary_term][:description])
+    @glossary_term = GlossaryTerm.new(
+      user: @user,
+      name: params[:glossary_term][:name],
+      description: params[:glossary_term][:description]
+    )
 
     if params[:glossary_term][:upload_image]
-      glossary_term.add_image(process_image(image_args))
+      @glossary_term.add_image(process_image(image_args))
     end
-    glossary_term.save
-    redirect_to(glossary_term_path(glossary_term.id))
+
+    return reload_form("new") unless @glossary_term.save
```

Comment:
There's a slight (7-year old) issue here, exacerbated by validation:
An image can be saved but the `@glossary_term.save` fails.
**Should I wrap them in a transaction?**
But I thinks that's not the right thing to do. In particular, if the image.save fails, then the term should nonetheless save (without the image).
Maybe (pseudocode
```
Save the term.
If that fails, 
  reload the form
else 
  something like`@glossary_term.thumb_image = image`
  redirect to the show page
end
```

Reply:
That won't work. If there's an image but no description, then the "save the term" will fail a validity check.